packages/didit-sdk/src/contexts/diditEmailAuthContext/DiditEmailAuthProvider.tsx

```diff
@@ -185,6 +173,27 @@ const DiditEmailAuthProvider = ({
     ]
   );
 
+  const loginWithSocial = useCallback(
+    (socialAuthProvider: SocialAuthProvider) => {
+      // Open a pop-up centered in the middle of the screen instead of redirecting
+      const width = DIDIT.EMAIL_AUTH_POPUP_WIDTH;
+      const height = DIDIT.EMAIL_AUTH_POPUP_HEIGHT;
+
+      // Recalculate the left and top positions just before opening the popup
+      const left = window.innerWidth / 2 - width / 2 + window.screenX;
+      const top = window.innerHeight / 2 - height / 2 + window.screenY;
+      var popupReference = window.open(
+        '',
+        'Authorization',
+        `width=${width},height=${height},left=${left},top=${top}`
+      );
+      if (popupReference) {
+        generateSocialAuthUrl(socialAuthProvider, popupReference);
```

Comment:
I like the refactor, but then `generateSocialAuthUrl` should only build the url, and do nothing with popup, that should be managed by this `loginWithSocialfunction`:
```suggestion
        const socialAuthUrl = generateSocialAuthUrl(socialAuthProvider);
        
        // Update popup url
        popupReference.location = authorizeUrl;
        setDiditEmailAuthPopup(popupReference);

        if (popup) startInitMessageInterval(popupReference);
```

 I think makes more sense and responsabilities are splited